rocket/src/TLB.scala

```diff
@@ -302,18 +301,43 @@ case class TLBConfig(
   * @param instruction true for ITLB, false for DTLB
   * @param lgMaxSize @todo seems granularity
   * @param cfg [[TLBConfig]]
-  * @param edge collect SoC metadata.
   */
-class TLB(instruction: Boolean, lgMaxSize: Int, cfg: TLBConfig)(implicit edge: TLEdgeOut, p: Parameters) extends CoreModule()(p) {
+class TLB(
+  instruction: Boolean,
+  lgMaxSize: Int,
+  cfg: TLBConfig,
+  memSlaves: Seq[MemSlaveParameters],
+  pmpGranularity: Int,
+  pgLevels: Int,
+  minPgLevels: Int,
+  pgLevelBits: Int,
+  pgIdxBits: Int,
+  vpnBits: Int,
+  ppnBits: Int,
+  vaddrBits: Int,
+  vaddrBitsExtended: Int,
+  paddrBits: Int,
+  hypervisorExtraAddrBits: Int,
+  asIdBits: Int,
+  xLen: Int,
+  cacheBlockBytes: Int,
+  memoryCacheable: Boolean,
+  memoryHomogenous: Boolean,
+  usingHypervisor: Boolean,
+  usingVM: Boolean,
+  usingAtomics: Boolean,
+  usingAtomicsInCache: Boolean,
+  usingAtomicsOnlyForIO: Boolean,
+  usingDataScratchpad: Boolean) extends Module {
   val io = IO(new Bundle {
     /** request from Core */
-    val req = Flipped(Decoupled(new TLBReq(lgMaxSize)))
+    val req = Flipped(Decoupled(new TLBReq(lgMaxSize, vaddrBitsExtended)))
     /** response to Core */
-    val resp = Output(new TLBResp())
+    val resp = Output(new TLBResp(paddrBits, vaddrBitsExtended))
     /** SFence Input */
-    val sfence = Flipped(Valid(new SFenceReq))
+    val sfence = Flipped(Valid((new SFenceReq(vaddrBits, asIdBits))))
     /** IO to PTW */
-    val ptw = new TLBPTWIO
+    val ptw = new TLBPTWIO() // TODO: Dependent on PTW
```

Comment:
I don't understand TODO here.
please don't use inline comment to make git diff happy